### ISSUE TITLE
Is "MockAjax is already installed" exception necessary?

### ISSUE THREAD

**Issue by aiankile** [jasmine/jasmine-ajax#186]
Currently, calling jasmine.Ajax.install() a second time before jasmine.Ajax.uninstall() unconditionally throws an exception ("MockAjax is already installed."). As consequence, if one test involving jasmine-ajax fails completely (that is, doesn't call afterEach / uninstall), all further tests automatically fail, as their beforeEach / install calls throw above exception.

Trying to understand the motivation for the current implementation: Why is using an already enabled jasmine-ajax considered a critical? Mainly due to possible garbage in stubs/requests not being cleaned up?

## Expected Behavior
jasmine.Ajax.install() should re-install and return silently if mockAjaxFunction is already bound to XMLHttpRequest.

## Current Behavior
"MockAjax is already installed." is thrown, breaking other tests.

## Possible Solution
a) call uninstall() implicitely, resetting internal vars, then return silently
b) optionally, introduce "strict" / "verbose" / "silent" mode or param for install(), affecting above behavior

## Your Environment
* Version used: 3.3.1
* Environment name and version: karma 2.0.0 / jasmine 2.9.1 / FF 58.0.2

**Comment by slackersoft** [comment]
This error is intended to alert the user that their testing environment is in an inconsistent state and the test suite may have test pollution.

If your call to `uninstall` is in an `afterEach` Jasmine should run it even if earlier parts of a spec fail. Unless that `afterEach` throws another error before making the call to `uninstall` you shouldn't have this problem.

I've updated the check and error message a bit to be less specific to Jasmine Ajax being installed already, instead checking if the `XMLHttpRequest` constructor has already been changed.

Hope this helps. Thanks for using Jasmine!
